app/indicators/service.py

```
@@ -1,2 +1,157 @@
+            'updated_at': str(r[8]) if r[8] else None,
+        }
+
+    def list_indicators(self, limit: int = 100, offset: int = 0) -> List[Dict[str, Any]]:
+        """
+        List indicators from the database with optional limit and offset parameters.
+        @param self - the instance of the class
+        @param limit - the maximum number of indicators to retrieve (default is 100)
+        @param offset - the offset from where to start retrieving indicators (default is 0)
+        @return a list of dictionaries containing indicator information
+        """
+        sel = f"SELECT id, indicator_name, category, description, formula, dependencies, parameters, created_at, updated_at FROM {self.TABLE} ORDER BY indicator_name LIMIT %(limit)s OFFSET %(offset)s"
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)